api/filters/authn/bearer.go

```diff
@@ -0,0 +1,61 @@
+package authn
+
+import (
+	"net/http"
+
+	"context"
+
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/security/oidc"
+)
+
+type bearerAuthnFilter struct {
+	Middleware
+}
+
+func NewBearerAuthnFilter(ctx context.Context, tokenIssuer, clientID string) (*bearerAuthnFilter, error) {
+	authenticator, err := oidc.NewAuthenticator(ctx, oidc.Options{
+		IssuerURL: tokenIssuer,
+		ClientID:  clientID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return &bearerAuthnFilter{
+		Middleware: Middleware{
+			authenticator: authenticator,
+			name:          "BearerAuthenticationFilter",
+		},
+	}, nil
+}
+
+func (ba *bearerAuthnFilter) Name() string {
+	return "BearerAuthenticationFilter"
+}
+
+func (ba *bearerAuthnFilter) FilterMatchers() []web.FilterMatcher {
```

Comment:
Let's suppose I want to reuse this Filter somewhere else? How can I do that as it is already bound to this specific Matchers?

Reply:
You would have to add more filterMatchers in this array (this array is kind of like an annotation - you "annotate" the filter with the places where it should run) - it works like the previous filter implementation - the only difference is  that before you could "annotate" a filter once (with one set of matchers) and now you can "annotate" it multiple times - with multiple sets fo matchers, meaning it can match to multiple endpoints.